src/api/ledger/orderbook.js

```diff
@@ -1,171 +1,54 @@
 'use strict';
 const _ = require('lodash');
+const async = require('async');
 const utils = require('./utils');
-const ripple = utils.common.core;
+const parseOrderbookOrder = require('./parse/orderbook-order');
 const validate = utils.common.validate;
-const bignum = require('bignumber.js');
+const composeAsync = utils.common.composeAsync;
+
+// account is to specify a "perspective", which affects which unfunded offers
+// are returned
+function getBookOffers(remote, account, ledgerVersion, limit,
+    takerGets, takerPays, callback) {
+  remote.requestBookOffers({
+    taker_gets: takerGets,
+    taker_pays: takerPays,
+    ledger: ledgerVersion || 'validated',
+    limit: limit,
+    taker: account
+  }, composeAsync((data) => data.offers.map(parseOrderbookOrder), callback));
+}
+
+function isSameIssue(a, b) {
+  return a.currency === b.currency && a.counterparty === b.counterparty;
+}
+
+function orderFilter(issue, direction, order) {
+  return isSameIssue(issue, order.specification.quantity)
+    && order.specification.direction === direction;
+}
+
+function formatBidsAndAsks(orderbook, orders) {
+  // the "base" currency is the currency that you are buying or selling
+  // the "counter" is the currency that the "base" is priced in
+  // a "bid"/"ask" is an order to buy/sell the base, respectively
+  const bids = orders.filter(_.partial(orderFilter, orderbook.base, 'buy'));
+  const asks = orders.filter(_.partial(orderFilter, orderbook.base, 'sell'));
```

Comment:
That's actually expected. The `orders` coming in are all bids and asks for both currencies, whereas the result we return is just the bids and asks for the base priced in terms of the counter. In other words, switching `base` and `counter` in the request will give you an entirely different set of orders.